src/twister.cpp

```diff
@@ -1581,21 +1615,43 @@ bool createSignedUserpost(entry &v, std::string const &username, int k,
         //userpost["t"] = "post";
         userpost["msg"] = msg;
     }
-    if ( rt != NULL && sig_rt != NULL ) {
+    switch(flag)
+    {
+    case USERPOST_FLAG_RT:
+    {
+        if (msg.size())
+        {
+            std::vector<char> buf;
+            bencode(std::back_inserter(buf), userpost);
+            std::string sig = createSignature(std::string(buf.data(),buf.size()), username);
+            if(sig.size())
+            {
+                v["sig_wort"] = sig;
```

Comment:
signature of post without rt and sig_rt. it's need to be able to drop them when you want to retwist some post with your comment which is retwist of some another post with comment to avoid overquoting.

e.g. post of `sometest`, k=66:

```
{
        "sig_userpost" : "20d6ef02cc111529b7c9fca9b4e3efcabe2c00cd5b6b6d77949b9296ae5213d78ffb0f577bc66033f0bd97cece538a02987ae2afe66379a6e62eab6ae3f7abdc96",
        "userpost" : {
            "height" : 98781,
            "k" : 66,
            "lastk" : 65,
            "msg" : "@sometest2 some?",
            "n" : "sometest",
            "rt" : {
                "height" : 98779,
                "k" : 33,
                "lastk" : 32,
                "msg" : "@sometest test?",
                "n" : "sometest2",
                "rt" : {
                    "height" : 77588,
                    "k" : 64,
                    "lastk" : 63,
                    "msg" : "test 30",
                    "n" : "sometest",
                    "time" : 1426006988
                },
                "sig_rt" : "\u001Fn?�E�gD�6#��;�*����\u0018\u000B�����\u000F\u001E��A.\nf/r���\u001C��2i�j�v�`���\u001D�/�s��\u0013�",
                "time" : 1439382741
            },
            "sig_rt" : "20f13e6a396eaf7b36cd18531c6f8aa712ff0461f07aee40c2fc9c5ce6a36e17f3965706378bd4c9495470af5468afff62fb685259337e832cc882f2ab5db91314",
            "time" : 1439383914
        }
    }
```

here's overquoting.

now see k=69-70:

```
{
        "sig_userpost" : "1fce56c36dbf1d3102a1b41b2db0f3b561902262564ffb5ad4be0de97a035a0b739dfdffef423e1aa8ac9db389b09d426bee78ce26e8551f48ad1dfc1c608e54c5",
        "sig_wort" : "20cb7bf8bc5a7994610ec7d4649db1d9e93050c64d69ee17a74e19d090d41e29ccb7b4fcb8a1f8a7e63d19b24d842f8c4391acac1e9882f6a5965285e89d68138d",
        "userpost" : {
            "height" : 98813,
            "k" : 70,
            "lastk" : 69,
            "msg" : "@sometest2 test!",
            "n" : "sometest",
            "rt" : {
                "height" : 98813,
                "k" : 69,
                "lastk" : 68,
                "msg" : "@sometest2 test?",
                "n" : "sometest",
                "time" : 1439408659
            },
            "sig_rt" : "1fd4026bc7dbe40eba92bd9d4f92eaba4333cfeedaa378621cec7eb9cd51ff5255349988357a13f7bc5ddd218f17d56c833dacfa142304b85d4f0dcd85b46f56a2",
            "time" : 1439409893
        }
    },
    {
        "sig_userpost" : "20c7017bfbd9df877bf0cc535c82b4a8bbb4fc7263c422929d362536fe3dd45bc5419a3e9ef1353ab3f7fdaffad209872f18b29a83c71ba66c5f8e161c37c6b624",
        "sig_wort" : "1fd4026bc7dbe40eba92bd9d4f92eaba4333cfeedaa378621cec7eb9cd51ff5255349988357a13f7bc5ddd218f17d56c833dacfa142304b85d4f0dcd85b46f56a2",
        "userpost" : {
            "height" : 98813,
            "k" : 69,
            "lastk" : 68,
            "msg" : "@sometest2 test?",
            "n" : "sometest",
            "rt" : {
                "height" : 77588,
                "k" : 30,
                "lastk" : 29,
                "msg" : "test",
                "n" : "sometest2",
                "time" : 1426006992
            },
            "sig_rt" : "203abab704f0b9e0d52585864202dc408ce6de2d49cccb4dc6c8c2fea71177140416ad21e4c45301b6ba483d617103e82dbfd08ce150801d6b4a9f9964dc81b68f",
            "time" : 1439408659
        }
    }
```